frontend/pages/shopping-lists/_id.vue

```diff
@@ -238,6 +239,12 @@ export default defineComponent({
       shoppingList.value = await fetchShoppingList();
     }
 
+    // constantly polls for changes
+    const pollTimer: ReturnType<typeof setInterval> = setInterval(() => { refresh() }, 5000);
+    onUnmounted(() => {
+      clearInterval(pollTimer);
+    });
+
```

Comment:
Looks good overall, can we wrap this in a Try/Catch and maybe cancel the interval if we make a few bad requests? If you went offline I wouldn't want it to just constantly keep polling. 

Also, should we consider a maximum amount of time to poll for? If I  left my browser open would I just want it to keep polling forever?

Reply:
> can we wrap this in a Try/Catch

For sure!

> should we consider a maximum amount of time to poll for?

Good point, any thoughts on how long? Maybe like 2 hours? I'm thinking that's the upper limit of how long it takes me to shop at one store. Otherwise, is there a way to capture focus? (like if there's no activity for 30 minutes or if it's not the active tab or something)

Reply:
There's the `isIdle` composable we could use

https://vueuse.org/core/useidle/#useidle

If we use that we could set a more aggressive timeout, like 5 minutes of idle and we stop polling, and re-trigger when idle is false again?

Reply:
Oh perfect that's exactly what I was looking for. Then yeah I think 5 minutes makes sense! If not we can always adjust later